### PR TITLE
Improve clr.Serialize

### PR DESCRIPTION
**Pull request by BCSharp**
Improvement on  #1361.

### CODE REVIEW

**Review by slozier — APPROVED** [review]

File: Src/IronPython/Runtime/ClrModule.cs [hunk 1]
```diff
@@ -1147,39 +1147,20 @@ private class PALHolder : MarshalByRefObject {
...
-
-                    data = self.ToString();
-                    format = type.FullName;
-                    break;
-
-                default:
-                    // something more complex, let the binary formatter handle it                    
-                    BinaryFormatter bf = new BinaryFormatter();
-                    MemoryStream stream = new MemoryStream();
+            if ((type.IsPrimitive && type.GetTypeCode() != TypeCode.Object)
```
Comment (slozier) [inline Src/IronPython/Runtime/ClrModule.cs:1150]: Are there primitive types with TypeCode.Object?

Comment (BCSharp) [reply Src/IronPython/Runtime/ClrModule.cs:1150]: `IntPtr` and `UIntPtr`

Comment (slozier) [reply Src/IronPython/Runtime/ClrModule.cs:1150]: Interesting, wonder why they made that choice...

Comment (BCSharp) [reply Src/IronPython/Runtime/ClrModule.cs:1150]: If I had to guess, perhaps because those types do not implement `IConvertible`.

**Pull request merged**